pallets/loans/src/entities/pricing/external.rs

```diff
@@ -150,9 +151,14 @@ impl<T: Config> ExternalActivePricing<T> {
 		}
 	}
 
-	fn linear_accrual_price(&self, maturity: Seconds) -> Result<T::Balance, DispatchError> {
+	fn linear_accrual_price(
+		&self,
+		maturity: Seconds,
+		price: T::Balance,
+		price_last_updated: Seconds,
+	) -> Result<T::Balance, DispatchError> {
 		Ok(cfg_utils::math::y_coord_in_rect(
-			(self.settlement_price_updated, self.latest_settlement_price),
+			(price_last_updated, price),
```

Comment:
This was wrong before. `price_last_updated` was in `millis` and maturity in `seconds`. I think maintaining everything as `<T as Config>::Moment` and just add `into_seconds()` here it should work.